pkg/costattribution/active_tracker.go

```diff
@@ -169,19 +228,42 @@ func (at *ActiveSeriesTracker) Collect(out chan<- prometheus.Metric) {
 	at.observedMtx.RLock()
 	if !at.overflowSince.IsZero() {
 		var activeSeries int64
+		var nativeHistogram int64
+		var nhBucketNum int64
 		for _, as := range at.observed {
-			activeSeries += as.Load()
+			activeSeries += as.activeSeries.Load()
+			nativeHistogram += as.nativeHistograms.Load()
+			nhBucketNum += as.nativeHistogramBuckets.Load()
 		}
 		at.observedMtx.RUnlock()
-		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+		out <- prometheus.MustNewConstMetric(at.activeSeriesPerUserAttribution, prometheus.GaugeValue, float64(activeSeries+at.overflowCounter.activeSeries.Load()), at.overflowLabels[:len(at.overflowLabels)-1]...)
+
+		if nhcounter := float64(nativeHistogram + at.overflowCounter.nativeHistograms.Load()); nhcounter > 0 {
+			out <- prometheus.MustNewConstMetric(at.activeNativeHistogramSeriesPerUserAttribution, prometheus.GaugeValue, nhcounter, at.overflowLabels[:len(at.overflowLabels)-1]...)
+			if bcounter := float64(nhBucketNum + at.overflowCounter.nativeHistogramBuckets.Load()); bcounter > 0 {
```

Comment:
Interestingly in the normal tracker+custom tracker, the two gauges are independently checked https://github.com/grafana/mimir/blob/bb31bb80cd62a5ef5525eebdd500c4cafa7b65f6/pkg/ingester/ingester.go#L828
Just for consistency I think we should do the same. It's easier to reason about if something goes wrong.